shared/actions/signup.js

```diff
@@ -1,366 +1,304 @@
 // @flow
 import logger from '../logger'
+import * as Constants from '../constants/signup'
 import * as LoginGen from './login-gen'
 import * as SignupGen from './signup-gen'
+import * as Saga from '../util/saga'
 import * as RPCTypes from '../constants/types/rpc-gen'
 import HiddenString from '../util/hidden-string'
 import {trim} from 'lodash-es'
 import {isMobile} from '../constants/platform'
 import {isValidEmail, isValidName, isValidUsername} from '../util/simple-validators'
 import {loginTab} from '../constants/tabs'
 import {navigateAppend, navigateTo} from '../actions/route-tree'
+import type {RPCError} from '../engine/types'
+import type {TypedState} from '../constants/reducer'
 
-function nextPhase() {
```

Comment:
remove this 'phase' concept. we don't do this anywhere else and it really mixed whats route state vs this internal state machine

Reply:
:tada: